nixarr/monitoring/default.nix

```diff
@@ -0,0 +1,263 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+with lib; let
+  cfg = config.nixarr;
+  
+  # Helper function to determine if a specific exporter should be enabled
+  shouldEnableExporter = service: 
+    cfg.${service}.enable && 
+    (cfg.monitoring.exporters.${service} or true);
+    
+  # Helper function to determine if a service is VPN-confined
+  isVpnConfined = service: cfg.${service}.enable && cfg.${service}.vpn.enable;
+    
+  # Helper function to create a script that extracts API key from config.xml
+  extractApiKeys = pkgs.writeShellApplication {
```

Comment:
I really like this, I do something similar for cross-seed, and I would like to merge the two somehow. I'll think about the simplest way to do so...

Maybe the API keys for all relevant services can be stored in something like `/data/.state/nixarr/api-keys`. Each file will then have permissions `sonarr:sonarr-api` and if a service needs access to the api file, it needs to be added to the group `sonarr-api`. The logic of extracting API keys should then be moved to something more general than exportarr of course.

Reply:
Yes, this is repetitive and probably required in more places in future. Let's do it.